ntbs-ui-tests/Hooks/DriverSetup.cs

```diff
@@ -38,8 +44,16 @@ public void BeforeScenario()
         }
 
         [AfterScenario]
-        public void AfterScenario()
+        public async Task AfterScenario()
         {
+            using (var connection = new SqlConnection(settings.EnvironmentConfig.ConnectionString))
+            {
+                connection.Open();
+                var deleteAlerts = "DELETE FROM Alert WHERE NotificationId IN @ids";
+                var deleteNotifications = "DELETE FROM Notification WHERE NotificationId IN @ids";
+                await connection.ExecuteAsync(deleteAlerts, new { ids = testContext.AddedNotificationIds.ToArray() });
+                await connection.ExecuteAsync(deleteNotifications, new { ids = testContext.AddedNotificationIds.ToArray() });
+            }
             Browser.Quit();
         }

```

Comment:
In some sense the trickiest bit of setting the tests up in this way is to ensure that they clean up after themselves properly. There are currently two different ways that `AddedNotificationIds` is populated:
- In the "Given I am on seeded notification..." step when we save a notification directly into the database
- In the "Then a new notification should have been created" step when we verify that a previous action has created a new notification.
It's all a bit fragile, because it's up to whoever is writing the tests to make sure that these are called appropriately. However, I couldn't think of a better way to do it.

Also, at the moment I've decided not to bother cleaning up the `AuditLogs` but I'm on the fence about this. It would be pretty easy to do.